subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ProjectDependenciesAttributesIntegrationTest.groovy

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.resolve.ResolveTestFixture
+import spock.lang.Unroll
+
+class ProjectDependenciesAttributesIntegrationTest extends AbstractIntegrationSpec {
+
+    ResolveTestFixture resolve = new ResolveTestFixture(buildFile, 'conf')
+
+    def setup() {
+        buildFile << """
+            configurations {
+               conf
+            }
+        """
+        settingsFile << """
+            rootProject.name = 'test'
+        """
+        resolve.prepare()
+    }
+
+    @Unroll
+    def "uses dependency attributes to select the right configuration on the target project (color=#color)"() {
+        given:
+        settingsFile << "include 'dep'"
+        buildFile << """
+            dependencies {
+                conf(project(':dep')) {
+                    attributes {
+                        attribute(Attribute.of('color', String), '$color')                        
+                    }
+                }
+            }
+        """
+        file("dep/build.gradle") << """
+            configurations {
+                blueVariant {
+                    canBeResolved = false
+                    canBeConsumed = true
+                    attributes {
+                        attribute(Attribute.of('color', String), 'blue')
+                    }
+                }
+                redVariant {
+                    canBeResolved = false
+                    canBeConsumed = true
+                    attributes {
+                        attribute(Attribute.of('color', String), 'red')
+                    }
+                }
```

Comment:
Is there a reason you chose to use separate configurations to define multiple variants instead of `configuration.outgoing.variants()`? Do you the 2 mechanisms resolve in the same way?

Reply:
Because that's the 1st thing that came to my mind. I probably need to add a test for the 2d option, because I don't think it works the same.